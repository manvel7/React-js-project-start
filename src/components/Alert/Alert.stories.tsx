import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Alert, {IProps} from "./Alert"

const meta: Meta<IProps> = {
  title: "Components/Alert",
  component: Alert,
  argTypes: {
    onClose: {action: "onClose"}
  }
}

const AlertStory: Story<IProps> = (args) => <Alert {...args}>Alert</Alert>

const DefaultDismissibleAlert = AlertStory.bind({})
DefaultDismissibleAlert.args = {
  isDismissible: true
}

const InfoAlert = AlertStory.bind({})
InfoAlert.args = {
  variant: "info"
}

const SuccessAlert = AlertStory.bind({})
SuccessAlert.args = {
  variant: "success"
}

const WarningAlert = AlertStory.bind({})
WarningAlert.args = {
  variant: "warning"
}

const ErrorAlert = AlertStory.bind({})
ErrorAlert.args = {
  variant: "error"
}

export {
  DefaultDismissibleAlert,
  InfoAlert,
  SuccessAlert,
  WarningAlert,
  ErrorAlert
}

export default meta
