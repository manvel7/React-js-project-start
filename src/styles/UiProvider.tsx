import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
