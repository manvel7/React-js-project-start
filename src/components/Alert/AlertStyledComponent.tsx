import {HTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"
import {h5Mixin} from "../../styles/typography"

export type AlertVariant = "primary" | "info" | "success" | "warning" | "error"

export interface IProps extends HTMLAttributes<HTMLElement> {
  isDismissible?: boolean
  variant?: AlertVariant
}

const primaryMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) => theme.palette.primary};
`

const infoMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) => theme.palette.alert.info};
`

const successMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) => theme.palette.alert.success};
`

const warningMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) => theme.palette.alert.warn};
`

const errorMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) => theme.palette.alert.error};
`

const AlertStyledComponent = styled.div<IProps>`
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: ${({theme}) => theme.palette.text.white};
  display: flex;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  ${h5Mixin<IProps>()}

  ${({variant}) => {
    switch (variant) {
      case "info":
        return infoMixin
      case "success":
        return successMixin
      case "warning":
        return warningMixin
      case "error":
        return errorMixin
      case "primary":
      default:
        return primaryMixin
    }
  }}
 
  &.hidden {
    opacity: 0;
    visibility: hidden;
  }
`

export default AlertStyledComponent
