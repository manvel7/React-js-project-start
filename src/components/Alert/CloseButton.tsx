import {ButtonHTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"
import {AlertVariant} from "./AlertStyledComponent"

export interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: AlertVariant
}

const primaryMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.primary};
`

const infoMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.info};
`

const successMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.success};
`

const warningMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.warn};
`

const errorMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.error};
`

const CloseButton = styled.button<IProps>`
  align-items: center;
  background-color: ${({theme}) => theme.palette.background.white};
  border-radius: 5rem;
  border-width: 0;
  color: ${({theme}) => theme.palette.primary};
  cursor: pointer;
  display: flex;
  height: 1.5rem;
  justify-content: center;
  margin-left: 1.5rem;
  width: 1.5rem;

  ${({variant}) => {
    switch (variant) {
      case "info":
        return infoMixin
      case "success":
        return successMixin
      case "warning":
        return warningMixin
      case "error":
        return errorMixin
      case "primary":
      default:
        return primaryMixin
    }
  }}
`

export default CloseButton
