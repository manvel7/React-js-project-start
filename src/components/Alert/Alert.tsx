import React, {FunctionComponent, HTMLAttributes, useState} from "react"
import {faTimes} from "@fortawesome/pro-solid-svg-icons"
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome"
import AlertStyledComponent, {AlertVariant} from "./AlertStyledComponent"
import CloseButton from "./CloseButton"

export interface IProps extends HTMLAttributes<HTMLDivElement> {
  isDismissible?: boolean
  variant?: AlertVariant
  onClose?: () => void
}

const Alert: FunctionComponent<IProps> = (props) => {
  const [dismissed, setDismissed] = useState(false)
  const {isDismissible, children, variant, onClose} = props

  const handleClose = () => {
    setDismissed(true)
    if (onClose) {
      onClose()
    }
  }

  return (
    <AlertStyledComponent
      role="alert"
      className={dismissed ? "hidden" : ""}
      isDismissible={isDismissible || false}
      variant={variant}>
      <div>{children}</div>
      {isDismissible ? (
        <CloseButton onClick={handleClose} variant={variant}>
          <FontAwesomeIcon icon={faTimes} />
        </CloseButton>
      ) : null}
    </AlertStyledComponent>
  )
}

export default Alert
