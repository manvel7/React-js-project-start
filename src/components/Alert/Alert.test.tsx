import React from "react"
import {render, screen} from "@testing-library/react"
import {UiProvider} from "../../styles"
import theme from "../../styles/theme"
import Alert from "./Alert"

describe("Alert", () => {
  describe("Default", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert>Default alert</Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
    it("dismissible should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert isDismissible={true}>Default dismissible alert</Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert>Default alert</Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.primary
      )
    })
    it("dismissible renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert isDismissible={true}>Default dismissible alert</Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const buttonElement = screen.getByRole("button")
      expect(buttonElement).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.primary
      )
    })
  })

  describe("Info", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="info">Info alert</Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
    it("dismissible should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="info" isDismissible={true}>
            Info dismissible alert
          </Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="info">Info alert</Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.info
      )
    })
    it("dismissible renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="info" isDismissible={true}>
            Info dismissible alert
          </Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const buttonElement = screen.getByRole("button")
      expect(buttonElement).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.info
      )
    })
  })

  describe("Success", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="success">Success alert</Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
    it("dismissible should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="success" isDismissible={true}>
            Success alert
          </Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="success">Success alert</Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.success
      )
    })
    it("dismissible renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="success" isDismissible={true}>
            Success dismissible alert
          </Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const buttonElement = screen.getByRole("button")
      expect(buttonElement).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.success
      )
    })
  })

  describe("Warning", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="warning">Warning alert</Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
    it("dismissible should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="warning" isDismissible={true}>
            Warning dismissible alert
          </Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="warning">Warning alert</Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.warn
      )
    })
    it("dismissible renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="warning" isDismissible={true}>
            Warning dismissible alert
          </Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const buttonElement = screen.getByRole("button")
      expect(buttonElement).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.warn
      )
    })
  })

  describe("Error", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="error">Error alert</Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
    it("dismissible should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Alert variant="error" isDismissible={true}>
            Error dismissible alert
          </Alert>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="error">Error alert</Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.error
      )
    })
    it("dismissible renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Alert variant="error" isDismissible={true}>
            Error dismissible alert
          </Alert>
        </UiProvider>
      )

      const element = screen.getByRole("alert")
      expect(element).toBeInTheDocument()

      const buttonElement = screen.getByRole("button")
      expect(buttonElement).toBeInTheDocument()

      const component = result.container.firstChild
      expect(component).toHaveStyleRule(
        "background-color",
        theme.palette.alert.error
      )
    })
  })
})
