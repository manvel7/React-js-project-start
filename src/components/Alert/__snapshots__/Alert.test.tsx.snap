// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Default dismissible should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border-radius: 5rem;
  border-width: 0;
  color: #001254;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 1.5rem;
  width: 1.5rem;
  color: #001254;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Default dismissible alert
  </div>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Alert Default should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Default alert
  </div>
</div>
`;

exports[`Alert Error dismissible should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #CA0000;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border-radius: 5rem;
  border-width: 0;
  color: #001254;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 1.5rem;
  width: 1.5rem;
  color: #CA0000;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Error dismissible alert
  </div>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Alert Error should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #CA0000;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Error alert
  </div>
</div>
`;

exports[`Alert Info dismissible should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border-radius: 5rem;
  border-width: 0;
  color: #001254;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 1.5rem;
  width: 1.5rem;
  color: #001254;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Info dismissible alert
  </div>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Alert Info should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #001254;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Info alert
  </div>
</div>
`;

exports[`Alert Success dismissible should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #0C5B0F;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border-radius: 5rem;
  border-width: 0;
  color: #001254;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 1.5rem;
  width: 1.5rem;
  color: #0C5B0F;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Success alert
  </div>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Alert Success should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #0C5B0F;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Success alert
  </div>
</div>
`;

exports[`Alert Warning dismissible should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #D4A600;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FEFEFE;
  border-radius: 5rem;
  border-width: 0;
  color: #001254;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 1.5rem;
  width: 1.5rem;
  color: #D4A600;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Warning dismissible alert
  </div>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`Alert Warning should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.5rem;
  bottom: 1.6rem;
  box-sizing: border-box;
  color: #FEFEFE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-transition: 0.6s linear all;
  transition: 0.6s linear all;
  visibility: visible;
  width: 100%;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 2.4rem;
  background-color: #D4A600;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0.hidden {
  opacity: 0;
  visibility: hidden;
}

<div
  class="c0"
  role="alert"
>
  <div>
    Warning alert
  </div>
</div>
`;
